terraform/modules/kubernetes/main.tf

```diff
@@ -0,0 +1,170 @@
+resource "kubernetes_deployment" "webapp" {
+  metadata {
+    name      = local.webapp_name
+    namespace = var.namespace
+  }
+  spec {
+    replicas = var.main_app.main["replicas"]
+    selector {
+      match_labels = {
+        app = local.webapp_name
+      }
+    }
+    template {
+      metadata {
+        labels = {
+          app = local.webapp_name
+        }
+      }
+      spec {
+        node_selector = {
+          "teacherservices.cloud/node_pool" = "applications"
+          "kubernetes.io/os"                = "linux"
+        }
+        container {
+          name    = local.webapp_name
+          image   = var.app_docker_image
+          command = try(slice(var.main_app.main["startup_command"], 0, 1), null)
+          args    = try(slice(var.main_app.main["startup_command"], 1, length(var.main_app.main["startup_command"])), null)
+
+          # Check performed to ensure the application is available. If it fails the current pod is killed and a new one created.
+          dynamic "liveness_probe" {
+            for_each = var.main_app.main["probe_path"]
+
+            content {
+              http_get {
+                path = join(",", var.main_app.main["probe_path"])
```

Comment:
As discussed, this can be replaced by `liveness_probe.value`

Reply:
changed var